source/views/sis/course-search/index.js

```diff
@@ -0,0 +1,226 @@
+// @flow
+
+import * as React from 'react'
+import {StyleSheet, View, Animated, Dimensions, Platform} from 'react-native'
+import {TabBarIcon} from '../../components/tabbar-icon'
+import * as c from '../../components/colors'
+import {CourseSearchBar} from '../components/searchbar'
+import {updateCourseData} from '../../../flux/parts/sis'
+import type {CourseType} from '../../../lib/course-search'
+import type {ReduxState} from '../../../flux'
+import type {TopLevelViewPropsType} from '../../types'
+import {connect} from 'react-redux'
+import groupBy from 'lodash/groupBy'
+import sortBy from 'lodash/sortBy'
+import toPairs from 'lodash/toPairs'
+import {CourseSearchResultsList} from './list'
+import LoadingView from '../../components/loading'
+
+export const deptNum = (course: CourseType) =>
+	`${course.departments.join('/')} ${course.number}${
+		course.section ? course.section : ''
+	}`
+
+type ReactProps = TopLevelViewPropsType
+
+type ReduxStateProps = {
+	allCourses: Array<CourseType>,
+	courseDataState: string,
+}
+
+type ReduxDispatchProps = {
+	updateCourseData: () => any,
+}
+
+type Props = ReactProps & ReduxStateProps & ReduxDispatchProps
+
+type State = {
+	searchResults: Array<{title: string, data: Array<CourseType>}>,
+	searchActive: boolean,
+	searchPerformed: boolean,
+}
+
+class CourseSearchView extends React.PureComponent<Props, State> {
+	static navigationOptions = {
+		tabBarLabel: 'Course Search',
+		tabBarIcon: TabBarIcon('search'),
+		title: 'SIS',
+	}
+
+	state = {
+		searchResults: [],
+		searchActive: false,
+		searchPerformed: false,
+	}
+
+	componentDidMount() {
+		this.props.updateCourseData()
+	}
+
+	animations = {
+		headerOpacity: {start: 1, end: 0, duration: 800},
+		searchBarTop: {start: 71, end: 10, duration: 800},
+		containerHeight: {start: 125, end: 64, duration: 800},
+	}
+
+	searchBar: any = null
+	headerOpacity = new Animated.Value(this.animations.headerOpacity.start)
+	searchBarTop = new Animated.Value(this.animations.searchBarTop.start)
+	containerHeight = new Animated.Value(this.animations.containerHeight.start)
+
+	performSearch = (text: string | Object) => {
+		const query = text.toLowerCase()
+		let results = this.props.allCourses.filter(course => {
+			return (
+				course.name.toLowerCase().includes(query) ||
+				(course.instructors || []).some(name =>
+					name.toLowerCase().includes(query),
```

Comment:
How useful will this be? Allowing search of a professor's name by anywhere in it, I mean? Should we make it start-only?

Reply:
This allows you to search by the professor's last name